service/src/main/java/uk/gov/hmcts/reform/fpl/service/ConfidentialDetailsService.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ConfidentialDetailsService {
+    private final ObjectMapper mapper;
+
+    public ConfidentialDetailsService(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public <T> List<Element<T>> addPartyMarkedConfidentialToList(Class<T> personType, List<Element<T>> details) {
+        List<Element<T>> confidentialList = new ArrayList<>();
+
+        details.stream()
+            .filter(element -> element.getValue() != null)
+            .forEach(value -> {
+                if (personType.isInstance(Respondent.class)) {
+                    if (respondentDetailsAreHidden(value)) {
+                        confidentialList.add(value);
+                    }
```

Comment:
Maybe both of them could implement confidential interface and that logic could be streamlined without reflection.